source/server/listener_manager_impl.cc

```diff
@@ -770,11 +784,19 @@ uint64_t ListenerManagerImpl::numConnections() {
 void ListenerManagerImpl::stopListener(Network::ListenerConfig& listener) {
   ENVOY_LOG(debug, "begin stop listener: name={}", listener.name());
   for (const auto& worker : workers_) {
-    worker->stopListener(listener);
+    if (!listenerStopped(listener.name())) {
+      stopped_listeners_.push_back(listener.name());
```

Comment:
This is nice future-proofing, but it doesn't work for the basic use case of "stop/drain all inbound listeners" (existing and future), we need a direction-based guard for that.

Reply:
the stopped_listeners_ will have all the inbound listeners of we call drain_listeners?inboundonly - would not that be sufficient?

Reply:
It will only have the inbound listeners that were active at the time of the operation, but LDS can add new inbound listeners afterwards, and if I'm reading this correctly, those listeners will be happily accepted by Envoy.

Reply:
I see.

Reply:
My mental model has been, no new ingress listeners will come up after pod/service has come up - they might change but not new ones.